strict-dependencies/index.js

```diff
@@ -87,7 +90,8 @@ module.exports = {
           dependency.allowReferenceFrom.some((allowPath) =>
             isMatch(relativeFilePath, allowPath),
           ) || // または同一モジュール間の参照が許可されている場合
-          (dependency.allowSameModule && isMatch(relativeFilePath, dependency.module))
+          (dependency.allowSameModule && isMatch(relativeFilePath, dependency.module)) ||
+            dependency.allowTypeImport && node.importKind === 'type'
```

Comment:
Including comments like this would be even better!

```suggestion
          (dependency.allowSameModule && isMatch(relativeFilePath, dependency.module))
          // または明示的に対象外としたtype importである場合
          || (dependency.allowTypeImport && node.importKind === 'type')
```